java-checks/src/main/java/org/sonar/java/checks/ConfigurationBeanNamesCheck.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S6862")
+public class ConfigurationBeanNamesCheck extends IssuableSubscriptionVisitor {
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return List.of(Tree.Kind.CLASS);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    var classTree = (ClassTree) tree;
+    if (!isConfigurationClass(classTree)) {
+      return;
+    }
+
+    var beanMethods = getBeanMethods(classTree);
+    var reportedElements = new HashSet<>();
+
+    for (int i = 0; i < beanMethods.size(); i++) {
+      String name = beanMethods.get(i).simpleName().name();
+      for (int j = i + 1; j < beanMethods.size(); j++) {
```

Comment:
Nested loop -> O(n²) complexity, and scanning a loop iteratively for the presence of an element (inner loop) is not a good practice. I would prefer the use of one loop and a HashSet instead:
```
var foundNames = new HashSet<String>();
for (MethodTree bean: beanMethods) {
  if (!foundNames.add(bean.simpleName().name()) {
    // report the issue here
  }
}
```